crates/forge/src/running.rs

```diff
@@ -68,17 +69,24 @@ fn build_hints_dict<'b>(
     (hints_dict, string_to_hint)
 }
 
-#[allow(clippy::too_many_arguments)]
 pub(crate) fn run_from_test_case(
     package_root: &Utf8PathBuf,
     runner: &SierraCasmRunner,
     case: &TestCase,
-    fork_targets: &[ForkTarget],
-    contracts: &HashMap<String, StarknetContractArtifacts>,
-    predeployed_contracts: &Utf8PathBuf,
     args: Vec<Felt252>,
-    environment_variables: &HashMap<String, String>,
+    runner_params: &RunnerParams,
+    runner_config: &RunnerConfig,
 ) -> Result<TestCaseSummary> {
+    // if only_ignored == true then tests without `#[ignore]` are filtered earlier
+    let should_be_run =
+        runner_config.include_ignored || runner_config.only_ignored || !case.ignored;
+
+    if !should_be_run {
+        return Ok(TestCaseSummary::Ignored {
+            name: case.name.clone(),
+        });
+    }
```

Comment:
This should be handler somewhere higher. If we enter `run_from_test_case` at all, the assumption should be that we do run the function.

Reply:
+1 We should just have a filter which passes a list of relevant test cases